### PR TITLE
Allow for creation of acls and http_access from class creation 

### PR DESCRIPTION
**Pull request by jcpunk**
Allow for creation of acls and http_access from class creation for hiera/ENC.  This should prevent the need to create a custom class to setup basic ACLs.  It should play nice with heira and ENC providers.

### CODE REVIEW

**Comment by jcpunk** [comment]
I fear I could use some help with tests for getting that behavior audited..... my attempts to write working ones failed.

File: manifests/config.pp [hunk 1]
```diff
@@ -7,6 +7,8 @@
   $coredump_dir                  = $squid::coredump_dir,
   $max_filedescriptors           = $squid::max_filedescriptors,
   $workers                       = $squid::workers,
+  $acls                          = $squid::params::acls,
```
Comment (traylenator) [inline manifests/config.pp]: Should be to init rather than params here.

File: manifests/params.pp [hunk 2]
```diff
@@ -8,4 +8,6 @@
   $coredump_dir                  = undef
   $max_filedescriptors           = undef
   $workers                       = undef
+  $acls                          = {}
```
Comment (traylenator) [inline manifests/params.pp]: ```undef``` is good here though you need to wrap the create_resources in an if statement.

File: README.md [hunk 3]
```diff
@@ -41,6 +41,8 @@ Parameters to the squid class almost map 1 to 1 to squid.conf parameters themsel
 * `coredump_dir` defaults to undef. [coredump_dir docs](http://www.squid-cache.org/Doc/config/coredump_dir/).
 * `max_filedescriptors` defaults to undef. [max_filedescriptors docs](http://www.squid-cache.org/Doc/config/max_filedescriptors/).
 * `workers` defaults to undef. [workers docs](http://www.squid-cache.org/Doc/config/workers/).
+* `acls` defaults to undef. [acl entries](http://www.squid-cache.org/Doc/config/acl/).
```
Comment (traylenator) [inline README.md]: I would maybe expand of the description of this argument as unlike the one to one mappings it is a lot less obvious. Maybe add an example or add to the existing example.

**Comment by jcpunk** [comment]
I think I've made the changes you requested and squashed them down to a single commit.

**Comment by traylenator** [comment]
The test fail is just syntax

```
Could not parse for environment *root*: Syntax error at 'validate_hash'; expected '}' at /home/travis/build/voxpupuli/puppet-squid/manifests/init.pp:35
```

a ```{``` is missing.

**Comment by jcpunk** [comment]
Weird, I'd swear I ran puppet parser validate against it...... (fixed)

**Comment by traylenator** [comment]
Please have a go at writing the tests. I'm happy to help but good for next time :-)

**Comment by jcpunk** [comment]
I'll see about getting to some trial bits soon.

**Comment by jcpunk** [comment]
I've taken a stab at it.  No idea why it works but copypasta with a bit of tweaking seems to have gotten me somewhere.

**Comment by traylenator** [comment]
Hi, tests are failing now for a general problem , should be better later today I hope.

**Comment by jcpunk** [comment]
I'll confess I have no idea what the errors there are trying to tell me.  If I need to do anything on my end, let me know.

**Comment by jcpunk** [comment]
Any chance for a new release with this patch?

**Pull request merged**